index.src.html

```diff
@@ -1380,6 +1380,28 @@ <h4 id="body-extraction" algorithm>
           6.  Set |Content-Type| to
               `application/x-www-form-urlencoded;charset=UTF-8`.
 
+          Otherwise, |data| is a generic object, so:
+
+          1.  If |data| has a property named like the |credential|'s
+              {{PasswordCredential/idName}} or
+              {{PasswordCredential/passwordName}}, <a>throw</a> a `TypeError`
+              exception.
```

Comment:
Dictionaries are [passed by value](https://heycam.github.io/webidl/#idl-dictionaries), and the [ES binding](https://heycam.github.io/webidl/#es-dictionary) describes pretty much the algorithm you suggest above, imho.

I'm relatively new to this, so take my advice with a grain of salt, but it seems that defining a dictionary instead of relying on object would fix your security concerns.